File: Arrays/2239. Find Closest Number to Zero/2239. Find Closest Number to Zero.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findClosestNumber = function (nums) {
  let c = nums[0];
  for (const n of nums) {
    if (Math.abs(n) < Math.abs(c) || (Math.abs(n) == Math.abs(c) && n > c)) {
      c = n;
    }
  }
  return c;
};

// Time: O(n)
// Space: O(1)

File: Arrays/2239. Find Closest Number to Zero/2239. Find Closest Number to Zero.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        c = nums[0]
        for n in nums:
            if abs(n) < abs(c) or abs(c) == abs(n) and n > c:
                c = n
        return c

# Time: O(n)
# Space: O(1)